svg.go

```diff
@@ -34,8 +35,21 @@ func processIcon(r io.Reader, w io.Writer) error {
 		if ok && tag.Name.Space == svgNamespace && tag.Name.Local == "svg" {
 			svgStartFound = true
 			depth++
-			if err := enc.EncodeToken(tok); err != nil {
-				return errgo.Notef(err, "cannot encode token %#v", tok)
+			st := tok.(xml.StartElement)
+			for i := range st.Attr {
+				if strings.ToLower(st.Attr[i].Name.Local) == "id" {
+					st.Attr = append(st.Attr[:i], st.Attr[i+1:]...)
```

Comment:
You might want to consider defining another function, say:

   // setAttr returns the given attributes with the given attribute name set to val,
   // adding an attribute if necessary.
   func setXMLAttr(attrs []xml.Attr, name xml.Name, val string) []xml.Attr

then the original function doesn't get grubbed up so much (and you can
reuse it elsewhere)